### PR TITLE
fixes #5788 - before_destroy missing from EnsureNotUsedBy.new in SmartProxy model

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/smart_proxy.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class SmartProxy < ActiveRecord::Base
   audited :allow_mass_assignment => true
 
   attr_accessible :name, :url, :location_ids, :organization_ids
-  EnsureNotUsedBy.new(:hosts, :hostgroups, :subnets, :domains, :puppet_ca_hosts, :puppet_ca_hostgroups)
+  before_destroy EnsureNotUsedBy.new(:hosts, :hostgroups, :subnets, :domains, :puppet_ca_hosts, :puppet_ca_hostgroups)
```
Comment (domcleal) [inline app/models/smart_proxy.rb]: realms too, come to think of it

Comment (isratrade) [inline app/models/smart_proxy.rb]: realm.rb has `before_destroy`

Comment (domcleal) [inline app/models/smart_proxy.rb]: right, but this line is checking that the proxy isn't used on another resource, so we should probably ensure it's not used by a realm too

Comment (isratrade) [inline app/models/smart_proxy.rb]: fixed and re-committed.

**Comment by domcleal** [comment]
Merged as f20ddfe for Foreman 1.5.1, thanks @isratrade.

**Pull request closed without merging**